### PR TITLE
Using `shlex.join()` when logging a command

### PR DESCRIPTION
**Pull request by dhermes**

### CODE REVIEW

File: nox/command.py [hunk 1]
```diff
@@ -84,7 +85,7 @@ def run(
         success_codes = [0]
 
     cmd, args = args[0], args[1:]
-    full_cmd = f"{cmd} {' '.join(args)}"
+    full_cmd = f"{cmd} {shlex.join(args)}"
```
Comment (dhermes) [inline nox/command.py:88]: Aw man; added in 3.8

Comment (dhermes) [reply nox/command.py:88]: Luckily the implementation is dead simple:

```python
def join(split_command):
    """Return a shell-escaped string from *split_command*."""
    return ' '.join(quote(arg) for arg in split_command)
```

File: nox/command.py [hunk 2]
```diff
@@ -67,6 +68,11 @@ def _clean_env(env: Optional[dict]) -> Optional[dict]:
     return clean_env
 
 
+def _shlex_join(args: Sequence[str]) -> str:
+    # shlex.join() was added in Python 3.8
+    return " ".join(shlex.quote(arg) for arg in split_command)
```
Comment (dhermes) [inline nox/command.py:73]: Whoops, copy-pasta life

File: nox/command.py [hunk 3]
```diff
@@ -84,7 +90,7 @@ def run(
         success_codes = [0]
 
     cmd, args = args[0], args[1:]
-    full_cmd = f"{cmd} {' '.join(args)}"
+    full_cmd = f"{cmd} {_shlex_join(args)}"
```
Comment (dhermes) [inline nox/command.py:93]: It occurs to me I should probably do a docs search for `python -m pip` and see what should be updated.

(As reviewers may already see from the branch name, I made this PR fully from the browser.)

**Review by theacodes — APPROVED** [review]

**Comment by theacodes** [comment]
Thanks @dhermes!

**Pull request merged**